src/Microsoft.TestPlatform.PlatformAbstractions/common/System/ProcessHelper.cs

```diff
@@ -50,8 +50,14 @@ public object LaunchProcess(string processPath, string arguments, string working
                     {
                         // Call WaitForExit again to ensure all streams are flushed
                         var p = sender as Process;
-                        p.WaitForExit();
-                        exitCallBack(p);
+                        try
+                        {
+                            p.WaitForExit();
+                            exitCallBack(p);
```

Comment:
In my opinion, we should not call exitCallBack if InvalidOperationException is thrown while accessing Process.HasExited API.

However, since we are also logging the testHostProcessStdError, I am moving this method call outside try catch block.